examples/advanced/quantum_volume.ipynb

```diff
@@ -98,23 +132,34 @@
     "fig, axs = plt.subplots()\n",
     "for idx, sampler in enumerate(samplers):\n",
     "    axs.scatter([d + idx / 10 for d in depths],\n",
-    "                sampler['probabilities'],\n",
-    "                marker=sampler['marker'],\n",
+    "                sampler['probabilities'].values(),\n",
+    "                marker='+',\n",
     "                c=f\"{sampler['color']}\",\n",
     "                label=f\"{sampler['label']}\")\n",
-    "# Line markers for asymptotic ideal heavy output probability and the ideal Heavy Output Generation threshold.\n",
+    "    axs.scatter([d + idx / 10 for d in depths],\n",
+    "                sampler['compiled-probabilities'].values(),\n",
+    "                marker=\"s\",\n",
+    "                c=f\"{sampler['color']}\",\n",
+    "                label=f\"{sampler['label']} Compiled\")\n",
+    "    axs.scatter([d + idx / 10 for d in depths],\n",
+    "                sampler['optimized-probabilities'].values(),\n",
+    "                marker=\"o\",\n",
+    "                c=f\"{sampler['color']}\",\n",
+    "                label=f\"{sampler['label']} Optimized\")\n",
```

Comment:
language is getting dicey: 

"optimized" vs. "compiled" is a distinction that isn't meaningful to me. A verbose but clear distinction could be between "routed" and "gate compiled".

Right now you've split up your "gate compiled" step into a pre-routing step and a post-routing step. I've been playing around with the code a bit and I found it to be quite nice to have the model circuit emit `TwoQubitMatrixGate`s instead of decomposed-into-cz. Then your gate compilation can happen all at once (after routing)

Reply:
Nice, love the `TwoQubitMatrixGate`s, the code is much simpler now!